cmake/enclave_cmake_wrappers.cmake

```diff
@@ -9,19 +9,18 @@ if (LVI_MITIGATION MATCHES ControlFlow)
     include(apply_lvi_mitigation)
 endif()
 
-# Wrapper of `add_executable`
-macro(add_enclave_executable NAME)
-    add_executable(${NAME} ${ARGN})
-    if (LVI_MITIGATION MATCHES ControlFlow)
+# Wrapper of `add_executable` for adding an enclave with LVI mitigation.
+macro(add_lvi_enclave_executable NAME)
+    if (LVI_MITIGATION MATCHES ControlFlow AND TARGET ${NAME})
         add_executable(${NAME}-lvi-cfg ${ARGN})
         apply_lvi_mitigation(${NAME}-lvi-cfg)
     endif()
-endmacro(add_enclave_executable)
+endmacro(add_lvi_enclave_executable)
 
 # Wrapper of `add_dependencies`
 macro(add_enclave_dependencies NAME)
     add_dependencies(${NAME} ${ARGN})
-    if (LVI_MITIGATION MATCHES ControlFlow)
+    if (LVI_MITIGATION MATCHES ControlFlow AND TARGET ${NAME}-lvi-cfg)
```

Comment:
Non-blocking: This is another example where it works, but intent is difficult to infer from the implementation, and it takes work on the part of the reader to reverse engineer:

* Why is it necessary to specify _both_ conditions? 
    * The presence of the LVI target should indicate that[ LVI_MITIGATION was specified, no?
    * Is there a condition we are concerned about when a previous build of that target is left over but the CMake configuration has changed? 
       * My understanding is that CMake does not permit this, but maybe this is a concern.
    * Alternatively, instead of using an artifact that implicitly encapsulates several other checks like the new flags introduced, is it a clearer expression of intent the wrap the multi-condition check in a macro/function and invoke that everywhere?
        * Another side effect to consider is this: is it desirable that a missing target causes the logic to be skipped with no errror (e.g. race condition in a build enviornment removes the lvi target while a build is running.)

Reply:
The side effect is exact the reason of adding this additional check.
For example  when `LVI_MITIGATION=ControlFlow` but `LVI_MITIGATION_SKIP_TESTS=ON`,
https://github.com/openenclave/openenclave/blob/cb79162e270da55a1cfa266f083719ac5038bd11/tests/ecall/enc/CMakeLists.txt#L15

Since the test is skipped, the enclave with lvi mitigation is not added. When calling the wrapper, cmake raises error of `the target does not exist` after passing the `if (LVI_MITIGATION MATCHES ControlFlow)` check.

I think check only the existence of the target should be sufficient. I'll update the PR accordingly.

Regarding the potential concern, I don't think will be a problem because once the CMake configuration is changed, CMake will re-initialize the dependency tree instead of using the cached one.